packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx

```diff
@@ -0,0 +1,252 @@
+        return theme.spacing(6);
+      default:
+        return 'auto';
+    }
+  }};
+  height: ${({ isMobile, theme }) =>
+    isMobile ? theme.modal.size.fullscreen : 'auto'};
+  max-height: ${({ isMobile }) => (isMobile ? 'none' : '90vh')};
+`;
+
+const StyledHeader = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 60px;
+  overflow: hidden;
+  padding: ${({ theme }) => theme.spacing(5)};
+`;
+
+const StyledContent = styled.div`
+  display: flex;
+  flex: 1;
+  flex: 1 1 0%;
+  flex-direction: column;
+  overflow-y: auto;
+  padding: ${({ theme }) => theme.spacing(10)};
+`;
+
+const StyledFooter = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 60px;
+  overflow: hidden;
+  padding: ${({ theme }) => theme.spacing(5)};
+`;
+
+const StyledBackDrop = styled(motion.div)<{
+  modalVariant: ModalVariants;
+}>`
+  align-items: center;
+  background: ${({ theme, modalVariant }) =>
+    modalVariant === 'primary'
+      ? theme.background.overlay
+      : theme.background.overlay2};
+  display: flex;
+  height: 100%;
+  justify-content: center;
+  left: 0;
+  position: fixed;
+  top: 0;
+  width: 100%;
+  z-index: 9999;
+  user-select: none;
+`;
+
+/**
+ * Modal components
+ */
+type ModalLayoutHeaderProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutHeader = ({ children, className }: ModalLayoutHeaderProps) => (
+  <StyledHeader className={className}>{children}</StyledHeader>
+);
+
+type ModalLayoutContentProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutContent = ({
+  children,
+  className,
+}: ModalLayoutContentProps) => (
+  <StyledContent className={className}>{children}</StyledContent>
+);
+
+type ModalLayoutFooterProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutFooter = ({ children, className }: ModalLayoutFooterProps) => (
+  <StyledFooter className={className}>{children}</StyledFooter>
+);
+
+/**
+ * Modal
+ */
+export type ModalSize = 'small' | 'medium' | 'large';
+export type ModalPadding = 'none' | 'small' | 'medium' | 'large';
+export type ModalVariants = 'primary' | 'secondary';
+
+export type EnhancedModalLayoutProps = React.PropsWithChildren & {
+  size?: ModalSize;
+  padding?: ModalPadding;
+  className?: string;
+  hotkeyScope?: ModalHotkeyScope;
+  onEnter?: () => void;
+  modalVariant?: ModalVariants;
+} & (
+    | { isClosable: true; onClose: () => void }
+    | { isClosable?: false; onClose?: never } // or the absence of `isClosable`
+  );
+
+const modalAnimation = {
+  hidden: { opacity: 0 },
+  visible: { opacity: 1 },
+  exit: { opacity: 0 },
+};
+
+// This component should be used over Modal when seeking a modal feel without modal state (hotkeyScope etc)
+export const EnhancedModalLayout = ({
+  children,
+  size = 'medium',
+  padding = 'medium',
+  className,
+  hotkeyScope = ModalHotkeyScope.Default,
+  onEnter,
+  isClosable = false,
```

Comment:
You say in the comment above the goal was to create a modal without hotkey scope, but you ended having "isClosable", "hotkeyScope" etc so I don't really understand the goal. Will keep reading more to understand

Reply:
That comment is a mistake, mb :(